src/Platform/Microsoft.Testing.Platform/Hosts/ServerTestHost.cs

```diff
@@ -93,13 +96,38 @@ public ServerTestHost(
 
     protected override bool RunTestApplicationLifeCycleCallbacks => true;
 
-    private void OnCurrentDomainUnhandledException(object sender, UnhandledExceptionEventArgs e)
-        => _logger.LogWarning($"[ServerTestHost.OnCurrentDomainUnhandledException] {e.ExceptionObject}{_environment.NewLine}IsTerminating: {e.IsTerminating}");
+    public string Uid => nameof(ServerTestHost);
 
-    private void OnTaskSchedulerUnobservedTaskException(object? sender, UnobservedTaskExceptionEventArgs e)
+    public string Version => AppVersion.DefaultSemVer;
+
+    public string DisplayName => PlatformResources.ServerTestHostDisplayName;
+
+    public string Description => PlatformResources.ServerTestHostDescription;
+
+#pragma warning disable VSTHRD100 // Avoid async void methods
+    private async void OnCurrentDomainUnhandledException(object sender, UnhandledExceptionEventArgs e)
+#pragma warning restore VSTHRD100 // Avoid async void methods
+    {
+        _logger.LogWarning($"[ServerTestHost.OnCurrentDomainUnhandledException] {e.ExceptionObject}{_environment.NewLine}IsTerminating: {e.IsTerminating}");
+
+        // Looks like nothing in this message to really be localized?
+        // All are class names, method names, property names, and placeholders. So none is localizable?
+        await ServiceProvider.GetOutputDevice().DisplayAsync(
```

Comment:
I think that use the `GetAwaiter().GetResult()` is better here. I don't expect a log of call here and awaiting a void we could lose the message.